### ISSUE TITLE
list index out of list

### ISSUE THREAD

**Issue by hugochugo** [bazil14/GTA_FIVEM_Modkit_extractor#1]
sorry if this is a newb question my first time using this and came up with this error
Traceback (most recent call last):
  File "extractmodids.py", line 25, in <module>
    mod = (kitnames[index].text)
IndexError: list index out of range

**Comment by bazil14** [comment]
no problem at all, this is my first time putting something out public so it's fun to get a bug report...

just before the error it would have printed into the console the meta file it was working on. would you be able to upload that file so I can have a look at it? something about it that the code doesn't like, I could do with some more error catching!

**Comment by hugochugo** [comment]
<?xml version="1.0" encoding="UTF-8"?>
<CVehicleModelInfoVarGlobal>    
  <Kits>
  <Lights>
	<Item>
      <id value="918" />	
      <indicator>
        <intensity value="0.375000" />
        <falloffMax value="2.500000" />
        <falloffExponent value="8.000000" />
        <innerConeAngle value="30.000000" />
        <outerConeAngle value="80.000000" />
        <emmissiveBoost value="false" />
        <color value="0xFFFF8000" />
      </indicator>
      <rearIndicatorCorona>
        <size value="0.000000" />
        <size_far value="0.000000" />
        <intensity value="0.000000" />
        <intensity_far value="0.000000" />
        <color value="0x00000000" />
        <numCoronas value="1" />
        <distBetweenCoronas value="128" />
        <distBetweenCoronas_far value="255" />
        <xRotation value="0.000000" />
        <yRotation value="0.000000" />
        <zRotation value="0.000000" />
        <zBias value="0.250000" />
        <pullCoronaIn value="false" />
      </rearIndicatorCorona>
      <frontIndicatorCorona>
        <size value="0.000000" />
        <size_far value="0.000000" />
        <intensity value="0.000000" />
        <intensity_far value="0.000000" />
        <color value="0x00000000" />
        <numCoronas value="1" />
        <distBetweenCoronas value="128" />
        <distBetweenCoronas_far value="255" />
        <xRotation value="0.000000" />
        <yRotation value="0.000000" />
        <zRotation value="0.000000" />
        <zBias value="0.250000" />
        <pullCoronaIn value="false" />
      </frontIndicatorCorona>
      <tailLight>
        <intensity value="0.030000" />
        <falloffMax value="4.000000" />
        <falloffExponent value="16.000000" />
        <innerConeAngle value="45.000000" />
        <outerConeAngle value="90.000000" />
        <emmissiveBoost value="false" />
        <color value="0xFFFF0000" />
      </tailLight>
      <tailLightCorona>
        <size value="0.050000" />
        <size_far value="0.100000" />
        <intensity value="0.100000" />
        <intensity_far value="0.100000" />
        <color value="0xFFFF0F05" />
        <numCoronas value="0" />
        <distBetweenCoronas value="0.1" />
        <distBetweenCoronas_far value="20.5" />
        <xRotation value="0.000000" />
        <yRotation value="0.000000" />
        <zRotation value="5.667000" />
        <zBias value="0.000000" />
        <pullCoronaIn value="false" />
      </tailLightCorona>
      <tailLightMiddleCorona>
        <size value="0.000000" />
        <size_far value="0.000000" />
        <intensity value="0.000000" />
        <intensity_far value="0.000000" />
        <color value="0x00000000" />
        <numCoronas value="0" />
        <distBetweenCoronas value="25" />
        <distBetweenCoronas_far value="25" />
        <xRotation value="0.000000" />
        <yRotation value="0.000000" />
        <zRotation value="0.000000" />
        <zBias value="0.000000" />
        <pullCoronaIn value="false" />
      </tailLightMiddleCorona>
      <headLight>
        <intensity value="1.000000" />
        <falloffMax value="35.000000" />
        <falloffExponent value="16.000000" />
        <innerConeAngle value="0.000000" />
        <outerConeAngle value="60.209999" />
        <emmissiveBoost value="false" />
        <color value="0xFFFFFFCC" />
        <textureName>VehicleLight_car_oldsquare</textureName>
        <mirrorTexture value="true" />
      </headLight>
      <headLightCorona>
        <size value="0.100000" />
        <size_far value="10.000000" />
        <intensity value="3.000000" />
        <intensity_far value="1.000000" />
        <color value="0xFFFFFFFF" />
        <numCoronas value="1" />
        <distBetweenCoronas value="128" />
        <distBetweenCoronas_far value="255" />
        <xRotation value="0.000000" />
        <yRotation value="0.000000" />
        <zRotation value="0.000000" />
        <zBias value="0.250000" />
        <pullCoronaIn value="false" />
      </headLightCorona>
      <reversingLight>
        <intensity value="0.100000" />
        <falloffMax value="4.000000" />
        <falloffExponent value="32.000000" />
        <innerConeAngle value="45.000000" />
        <outerConeAngle value="90.000000" />
        <emmissiveBoost value="false" />
        <color value="0xFFFFFFFF" />
      </reversingLight>
      <reversingLightCorona>
        <size value="0.100000" />
        <size_far value="0.100000" />
        <intensity value="0.100000" />
        <intensity_far value="0.400000" />
        <color value="0x00F7F7F7" />
        <numCoronas value="0" />
        <distBetweenCoronas value="128" />
        <distBetweenCoronas_far value="255" />
        <xRotation value="0.000000" />
        <yRotation value="0.000000" />
        <zRotation value="0.000000" />
        <zBias value="0.250000" />
        <pullCoronaIn value="false" />
      </reversingLightCorona>
      <name>denalihd</name>
    </Item>
  </Lights>
</CVehicleModelInfoVarGlobal>

thanks for the quick response really excited about this

**Comment by hugochugo** [comment]
how do i send the whole thing?

**Comment by hugochugo** [comment]
https://filebin.net/a8boerl87fktx7xr
seems to not have a modkit section could that be why

**Comment by hugochugo** [comment]
alright so i took that carcols out for now untill i fix it and it ran all the way through and made the csv what am I supposed to use to open it. I opened it with xls worksheet and im kind of confused with the results. Again sorry for the newb questions

**Comment by bazil14** [comment]
that could be why, can you run the file though an online XML checker? there seems to be a mismatch of tags, I'll look into adding some code tomorrow so the program isn't tripped up by it anymore and instead reports it as something to be looked at.

**Comment by hugochugo** [comment]
it's all good now think just the carcol was messed up on that one car will redo it. As for the results should I be fixing the duplicate entries?

**Comment by bazil14** [comment]
open it up with wordpad or notepad ++, though excel should work too iirc.  its just a file to allow you to spot any clashes with your existing modkit id's as well as see the free spaces of where you can move them. dont move modkit ids above 1000 or below around 350ish (will clash with the standard cars)

I created this tool as the server I run for some friends has around 600 cars and managing the mod kit id's is a pain.  I might add an auto id moving feature in the future  


the format comes out as binary, ID, modkit name

**Comment by hugochugo** [comment]
yes i can see the stress as we only have about 350 and its already becoming a hassle. so when i seen this i got a excited

**Comment by bazil14** [comment]
yeah fix the duplicates :) , there is no guarantee it will fix all the modkits, GTA does some weird internal maths on the IDs that no one truly understands, if something isn't working and isn't clashing with another car, try moving it around to a different id

**Comment by hugochugo** [comment]
i have some modkit ids under 350 but it looks like most of them are vanilla cars should i still move them above 350?

**Comment by hugochugo** [comment]
https://filebin.net/u8wi4avwsuk6yl5i
as you can see it looks like i have alot of work coming my way lol

**Comment by bazil14** [comment]
I think you can leave them be, I don't know where the standard cars come up to ID wise at the moment, it's just something to be aware of as the tool will present this as a big blank area in the low numbers  which looks temping to put cars into, but you shouldn't haha.

**Comment by bazil14** [comment]
I would say most of those low ones are default modkits so can be ignored, but yeah 😅, it's a pain! also one thing to be aware of is the order of how the meta data files load is important to get working modkits too. I have written a tool to prepare cars for fivem on bulk that I may be releasing sometime soon. anyway, I best sleep!

**Comment by hugochugo** [comment]
have a good night thanks for this
